Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -315,6 +325,9 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
     ///   - bundle: The bundle that was removed.
     public func dataProvider(_ dataProvider: DocumentationContextDataProvider, didRemoveBundle bundle: DocumentationBundle) throws {
         documentationCacheBasedLinkResolver.unregisterBundle(identifier: bundle.identifier)
+        if LinkResolutionMigrationConfiguration.shouldSetUpHierarchyBasedLinkResolver {
+            hierarchyBasedLinkResolver?.unregisterBundle(identifier: bundle.identifier)
```

Comment:
If `shouldSetUpHierarchyBasedLinkResolver` is `false`, wouldn't `hierarchyBasedLinkResolver` always be `nil`? Do we need the conditional?

Reply:
Correct. I've updated this code.